frontend/src/utils/time.ts

```diff
@@ -19,8 +19,25 @@ import { usei18n } from '@/composables';
  * Get dateFns locale
  */
 function getDateFnsLocale(locale: string): Locale | undefined {
-  //@ts-expect-error - Some of our locales are not present in datefns.
-  return datefnslocales[locale];
+  if (datefnslocales[locale as keyof typeof import('date-fns/locale')]) {
+    return datefnslocales[locale as keyof typeof import('date-fns/locale')];
+  } else if (
+    datefnslocales[
+      locale.replace('-', '') as keyof typeof import('date-fns/locale')
+    ]
+  ) {
+    return datefnslocales[
+      locale.replace('-', '') as keyof typeof import('date-fns/locale')
+    ];
+  } else if (
+    datefnslocales[
+      locale.split('-')[0] as keyof typeof import('date-fns/locale')
+    ]
+  ) {
+    return datefnslocales[
+      locale.split('-')[0] as keyof typeof import('date-fns/locale')
+    ];
+  }
```

Comment:
Why we need this?
Please also import the type at the top-level or using ES6 import syntax, don't need to duplicate the import literal this many times.

Reply:
We need this due to the differences between the vue i18n and date-fns locales.

* 1st case: when there is a date-fns locale equals to the i18n, it's all good
* 2nd case: vue i18n got locales such as `fr-FR` but date-fns got them as `frFR`. That's to match those
* 3rd case: if date-fns doesn't have the `frFR` locale, it's to check if `fr` exists

Without it, we strongly limit those matching

Reply:
Copy and paste this into the code comments please